zulip/zulip/__init__.py

```diff
@@ -1728,3 +1741,51 @@ def hash_util_decode(string: str) -> str:
     # Acknowledge custom string replacements in zulip/zulip's zerver/lib/url_encoding.py before unquoting.
     # NOTE: urllib.parse.unquote already does .replace('%2E', '.').
     return urllib.parse.unquote(string.replace(".", "%"))
+
+
+# Will be set to True during server CI tests for zulip of version 3.2 or prior to that
+USE_LEGACY_INTERFACE = os.environ.get("USE_LEGACY_INTERFACE", "False").lower() == "true"
+
+
+class LegacyInterfaceClient(Client):
+    """
+    This is a hack that we use to forcibly override the function signature of certain API bindings to make them support
+    older versions of zulip server in the CI.
+    """
+
+    @overload
+    def update_user_group_members(
+        self, user_group_id: int, group_data: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        ...
+
+    @overload
+    def update_user_group_members(self, group_data: Dict[str, Any]) -> Dict[str, Any]:
+        ...
+
+    def update_user_group_members(self, *args: Any, **kwargs: Any) -> Dict[str, Any]:
```

Comment:
There is no reason to use an overload here, and even if you did need to use an overload, that would not be a reason to use the wildly type-unsafe `*args: Any, **kwargs: Any` signature.

```python
    def update_user_group_members(
        self, user_group_id: Optional[int] = None, group_data: Dict[str, Any]
    ) -> Dict[str, Any]:
```